web/source/keyboards/kmwkeyboards.ts

```diff
@@ -477,7 +477,12 @@ namespace com.keyman.keyboards {
       // if(!PLgCode && (<KeymanBase>keymanweb).keyboardManager.activeStub) {
       //   PLgCode = (<KeymanBase>keymanweb).keyboardManager.activeStub['KLC'];
       // }
-      this.doKeyboardChange(PInternalName, PLgCode);
+      let _this = this;
+      p.then(function() {
+        // Only mark the keyboard as having changed once the setActiveKeyboard op
+        // is successful.
+        _this.doKeyboardChange(PInternalName, PLgCode);
+      })
```

Comment:
```suggestion
      p.then(function() {
        // Only mark the keyboard as having changed once the setActiveKeyboard op
        // is successful.
        _this.doKeyboardChange(PInternalName, PLgCode);
      });
```

Is there a difference with `this` if we use an arrow function here? (Something that's still a little fuzzy in my mind)

Reply:
The issue is that within the anonymous function, `this` refers to the Promise that will be resolving, if my understanding is correct - not the current class.  Hence `const _this = this;` beforehand.

Reply:
The reason I asked about arrow functions is that [they inherit `this` from the lexical scope](https://www.codementor.io/@dariogarciamoya/understanding-this-in-javascript-with-arrow-functions-gcpjwfyuc) (so much more sensibler). [more yada-yada that may be relevant](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/Arrow_functions)!

Reply:
Oh, interesting.  I do remember issues with TS arrow-function compilation, though - I feel like they had an issue with certain older browsers we still aim to support.  I can't point to anything specific, though.